content/docs/intro/concepts/config.md

```diff
@@ -430,3 +430,61 @@ For example, this configures a stack to use a HashiCorp Vault transit key named
 $ pulumi stack init my-stack \
     --secrets-provider="hashivault://payroll"
 ```
+
+### Changing the Secrets Provider for a Stack
+
+> Note: There is not (yet) built-in `pulumi` CLI support for changing the secrets provider on an existing stack, but for cases where this is necessary, it is possible with some manual steps, as outlined in this section. 
+
+> Note: These steps can also be used to change the passphrase that is used for an existing stack.
+
+There are two places where secrets are stored in an encrypted form: (1) the configuration stored in the stack file (`Pulumi.<stackname>.yaml`) and (2) the state stored in your state backend.  To change the secrets provider for a stack, both of these will need to be manually updated.
+
+First, you will need to get copies of the two assets above with plaintext copies of your secret values from your current stack.  This is a sensitive operation, and you should be careful to do this in a safe environment and be sure to remove these files when you are done.
+
+```
+$ pulumi config --show-secrets --json > config.json
+$ pulumi stack export --show-secrets > stack.json
+```
+
+Next, you will need to create a temporary new stack using the desired new secrets provider.  If using the `passphrase` secrets provider, this can also be used to create a new passphrase to use for encryption.
+
+```
+$ pulumi stack init temp --secrets-provider <provider>
+```
+
+If you chose `passphrase` you will see something like this:
+
+```
+$ cat Pulumi.temp.yaml
+encryptionsalt: v1:h8kauiCOXZc=:v1:0Tjy4uHvwgw/b0Hp:gKAqRJ74vyIkc5BeEUlCFXynwRTPFQ==
+```
+
+Else if you chose one of the cloud-provider secrets providers you will see something like this:
+
+```
+$ cat Pulumi.temp.yaml
+secretsprovider: awskms://alias/my-kms-key
+encryptedkey: AQICAHhivZtfYgNwNJDVWDd++88uo2qgy5MJqqzsPfTFv7uzDgF+1kaw9/wSKgGh5QpT/cwtAAAAfjB8BgkqhkiG9w0BBwagbzBtAgEAMGgGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMc2EiH7Mli5ohzB1aAgEQgDsOszkuuC7wffN3FdtKCNAcRoJq3krWmeA/ucnSMn/F2OYVIDKxDZkjJYZBAjEmpWL49Fcly/n8NMNHhw==
+```
+
+And if you chose the Pulumi Service secrets provider, there will be no stack file.
+
+You will use the values above to replace those in use in your existing stack.  
+
+First, your config.  
```

Comment:
Roughly here is where I think we need the `pulumi stack select ...` back to the original stack as I mentioned in https://github.com/pulumi/docs/pull/3342/files#r428351515.